builtin/providers/azurerm/resource_arm_virtual_network.go

```diff
@@ -257,16 +278,55 @@ func getVirtualNetworkProperties(d *schema.ResourceData) *network.VirtualNetwork
 			DNSServers: &dnses,
 		},
 		Subnets: &subnets,
-	}
+	}, nil
 }
 
 func resourceAzureSubnetHash(v interface{}) int {
+	var buf bytes.Buffer
 	m := v.(map[string]interface{})
-	subnet := m["name"].(string) + m["address_prefix"].(string)
-	if securityGroup, present := m["security_group"]; present {
-		subnet = subnet + securityGroup.(string)
+	buf.WriteString(fmt.Sprintf("%s-", m["name"].(string)))
+	buf.WriteString(fmt.Sprintf("%s-", m["address_prefix"].(string)))
+	if m["security_group"] != nil {
+		buf.WriteString(fmt.Sprintf("%s-", m["security_group"].(string)))
+	}
+	return hashcode.String(buf.String())
```

Comment:
given a `-` has been added at the end of each value this hash code is going to be different, and thus a change will show up as required on the plan. To avoid this, we can either [increment the Schema Version and write a State Migration](XXX) - or just remove the `-` from these strings (which I would suggest here :))